### ISSUE TITLE
hi, why the scores are sometimes very lower which might be easily ignored

### ISSUE THREAD

**Issue by jinfagang** [IDEA-Research/detrex#76]
I visualized some images using DAB_DETR resnet50 pretrained model:

<img width="880" alt="image" src="https://user-images.githubusercontent.com/21303438/191956690-930f1850-2206-4eef-a1d7-98b60e602026.png">

<img width="752" alt="image" src="https://user-images.githubusercontent.com/21303438/191956741-e5c90332-362e-4080-b227-d47a29333306.png">

<img width="708" alt="image" src="https://user-images.githubusercontent.com/21303438/191956802-ecb097a6-390e-488f-8a49-373520d2ba03.png">

so as for the score trending be lower, if a threshold set too low, will have many false postives, if set to high, easily miss important objects.

How to make the score confidence  more reliable ?

**Comment by SlongLiu** [comment]
Thanks for providing the question. The reason might come from the focal loss for classification. We mentioned this in many DETR like models. We suggest using low thersholds in these models, like 0.2-0.3. By

**Comment by jinfagang** [comment]
Setting  a lower thresh might introduce some unwanted false positives, does there any hack on this? From practical of view, lower score caused too many issues in real world applications.

**Comment by SlongLiu** [comment]
I think it is a common tradeoff between false positive and false negative results. Some other models may suffer from the problem as well.

In our experiments, with a lower threshold like 0.2-0.3, the predicted results are still meaningful. For example, the model might output some objects omitted in the ground truth. It rarely gives predictions that are totally wrong. A well-trained model causes nearly no real false positives. Hence, we think a lower threshold is acceptable.

**Comment by SlongLiu** [comment]
One way to alleviate this is to rank the predictions category by category and then select predictions based on each category's mean and var of prediction scores. Some heuristic ways might be helpful as well.

I think it remains a good question for the community to explore, as the problem is not faced by the DAB-DETR only but all the focal loss based algorithms.

**Comment by jinfagang** [comment]
@SlongLiu thank u! focal loss indeed caused too lower scores.
